analyzer/decrypt_crypto.cc

```diff
@@ -167,23 +171,19 @@ DecryptionInformation remove_header_protection(std::vector<uint8_t> client_hp,
 		decoded_packet_number = unprotected_header[encrypted_offset + i] |
 		                        (decoded_packet_number << 8);
 		}
-	std::vector<uint8_t> protected_header(encrypted_packet.begin(),
-	                                      encrypted_packet.begin() + encrypted_offset +
-	                                          recovered_packet_number_length);
 
 	// Store the information back in the struct
 	decryptInfo.packet_number = decoded_packet_number;
 	decryptInfo.packet_number_length = recovered_packet_number_length;
-	decryptInfo.protected_header = protected_header;
-	decryptInfo.unprotected_header = unprotected_header;
+	decryptInfo.unprotected_header = std::move(unprotected_header);
 	return decryptInfo;
 	}
 
 /*
 Calculate the nonce for the AEAD by XOR'ing the CLIENT_IV and the
 decoded packet number, and returns the nonce
 */
-std::vector<uint8_t> calculate_nonce(std::vector<uint8_t> client_iv, uint64_t packet_number)
+std::vector<uint8_t> calculate_nonce(const std::vector<uint8_t>& client_iv, uint64_t packet_number)
```

Comment:
Unless microbenchmarks show a clear benefit, I'd instead keep the pass by value here and remove the copy in the body.